lib/redlock/client.rb

```diff
@@ -228,6 +258,22 @@ def lock_instances(resource, ttl, options)
       end
     end
 
+    def try_get_remaining_ttl(resource, value)
+      ttls, time_elapsed = timed do
+        @servers.map { |s| s.get_remaining_ttl resource, value }.compact
+      end
+
+      if ttls.size >= @quorum
+        min_ttl = ttls.sort.last(@quorum).min
+        min_ttl - time_elapsed - drift(min_ttl)
+      else
+        # TODO: This nil is overloaded, it could mean that we did not find the
+        # value, or that we could not achieve quorum. Maybe we want to do
+        # something different in the latter case?
+        nil
```

Comment:
IMO this is ok, and certainly so as the interface of the `get_remaining_ttl()` function. 

> it could mean that we did not find the value, or that we could not achieve quorum

isn't this the same case? at least from the user's perspective?

Reply:
Hmm, I guess I'm worried about a situation where we could only reach say 2/5 servers, and so we really don't know the TTL. Now that I think about it, assuming the lock has expired should be fine behaviour from the users perspective,